api/src/hyp3_api/validation.py

```diff
@@ -0,0 +1,73 @@
+import json
+import os
+
+import requests
+from shapely.geometry import Polygon, shape
+
+from hyp3_api import CMR_URL
+
+
+class GranuleValidationError(Exception):
+    pass
+
+
+def validate_granules(granules):
+    granule_metadata = get_cmr_metadata(granules)
+
+    check_granules_exist(granules, granule_metadata)
+    check_dem_coverage(granule_metadata)
+
+
+def get_cmr_metadata(granules):
+    cmr_parameters = {
+        'producer_granule_id': granules,
+        'provider': 'ASF',
+        'short_name': [
+            'SENTINEL-1A_SLC',
+            'SENTINEL-1B_SLC',
+        ],
+    }
+    response = requests.post(CMR_URL, data=cmr_parameters)
+    response.raise_for_status()
+    granules = [
+        {
+            'name': entry['producer_granule_id'],
+            'polygon': Polygon(format_points(entry['polygons'][0][0]))
+        } for entry in response.json()['feed']['entry']
+    ]
+    return granules
+
+
+def check_granules_exist(granules, granule_metadata):
+    found_granules = [granule['name'] for granule in granule_metadata]
+    not_found_granules = set(granules) - set(found_granules)
+    if not_found_granules:
+        raise GranuleValidationError(f'Some requested scenes could not be found: {",".join(not_found_granules)}')
+
+
+def check_dem_coverage(granule_metadata):
+    coverage = get_coverage_shapes_from_geojson()
+    bad_granules = [granule['name'] for granule in granule_metadata if
+                    not check_intersects_with_coverage(granule['polygon'], coverage)]
```

Comment:
probably make `bad_granules` a set instead of a list to eliminate duplicates in the error message